### PR TITLE
Fix tablefunction not support expr

### PR DESCRIPTION
**Pull request by mooleetzi**
## What type of PR is this?

- [ ] API-change
- [X] BUG
- [ ] Improvement
- [ ] Documentation
- [X] Feature
- [ ] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue #5626, #5562

## What this PR does / why we need it:

1. extend json_extract.

2. make tablefunction generate_series,unnest support normal expr

### CODE REVIEW

File: pkg/sql/colexec/unnest/unnest.go [hunk 1]
```diff
@@ -63,30 +60,109 @@ func Prepare(_ *process.Process, arg any) error {
 
 func Call(_ int, proc *process.Process, arg any) (bool, error) {
 	param := arg.(*Argument).Es
-	if param.isCol {
+	switch param.typ {
+	case "str":
```
Comment (fengttt) [inline pkg/sql/colexec/unnest/unnest.go:64]: This has a very bad taste -- I wonder WHY should you care about param type?   It is an expression and get evaluated, shouldn't this be the same?

Yuanning, Long ran or Ming Song/Mo Chen, one of them must have a strong opinion on this.

**Review by fengttt — CHANGES_REQUESTED** [review]

**Review by aressu1985 — APPROVED** [review]

**Review by fengttt — APPROVED** [review]
Please try to break PR into smaller, separate pieces.   Thank you.l

**Comment by mooleetzi** [comment]
> Please try to break PR into smaller, separate pieces. Thank you.l

ok

**Review by nnsgmsone — APPROVED** [review]

**Review by iamlinjunhong — APPROVED** [review]

**Review by ouyuanning — APPROVED** [review]

File: pkg/sql/colexec/expr_eval.go [hunk 2]
```diff
@@ -85,7 +85,7 @@ func getConstVec(proc *process.Process, expr *plan.Expr, length int) (*vector.Ve
 		case *plan.Const_Timeval:
 			vec = vector.NewConstFixed(constTimeType, length, t.C.GetTimeval(), proc.Mp())
```
[the remark this answers is not in the document]
Comment (mooleetzi) [reply pkg/sql/colexec/expr_eval.go:86]: Ok,I will check.

**Review by aunjgr — APPROVED** [review]

**Pull request merged**